src/AspNet.Security.OAuth.EVEOnline/EVEOnlineAuthenticationHandler.cs

```diff
@@ -20,44 +22,81 @@ namespace AspNet.Security.OAuth.EVEOnline
 {
     public class EVEOnlineAuthenticationHandler : OAuthHandler<EVEOnlineAuthenticationOptions>
     {
+        private readonly JwtSecurityTokenHandler _tokenHandler;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EVEOnlineAuthenticationHandler"/> class.
+        /// </summary>
+        /// <param name="options">The authentication options.</param>
+        /// <param name="logger">The logger to use.</param>
+        /// <param name="encoder">The URL encoder to use.</param>
+        /// <param name="clock">The system clock to use.</param>
+        /// <param name="tokenHandler">The JWT security token handler to use.</param>
         public EVEOnlineAuthenticationHandler(
             [NotNull] IOptionsMonitor<EVEOnlineAuthenticationOptions> options,
             [NotNull] ILoggerFactory logger,
             [NotNull] UrlEncoder encoder,
-            [NotNull] ISystemClock clock)
+            [NotNull] ISystemClock clock,
+            [NotNull] JwtSecurityTokenHandler tokenHandler)
             : base(options, logger, encoder, clock)
         {
+            _tokenHandler = tokenHandler;
         }
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync(
             [NotNull] ClaimsIdentity identity,
             [NotNull] AuthenticationProperties properties,
             [NotNull] OAuthTokenResponse tokens)
         {
-            using var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint);
-            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
-            request.Headers.Authorization = new AuthenticationHeaderValue("Bearer", tokens.AccessToken);
+            // JwtSecurityToken decodedToken = new JwtSecurityToken(tokens.AccessToken);
+            var tokenClaims = ExtractClaimsFromToken(tokens.AccessToken);
```

Comment:
Whattttttttttt? Reading and assuming access tokens have a specific format - JWT here - in OAuth 2.0 client is a horrible practice. Is this really the pattern they are recommending now?

Reply:
I am not sure what you mean here?

In terms of the AccessToken being a JWT that is what the devs are saying it is.
Documented [https://docs.esi.evetech.net/docs/sso/web_based_sso_flow.html](https://docs.esi.evetech.net/docs/sso/web_based_sso_flow.html)

Reply:
Access tokens are opaque for the clients, that are not supposed to read them: they are only meant to be read by the resource APIs they target. Doing that is a serious protocol violation (that's why we have a different token called `id_token` in OpenID Connect).

Do we know when the v1 endpoints will be deprecated? I'd prefer if we waited a bit before adopting this insane non-standard approach.

Reply:
Ok, thats fair enough.

We have no set day, all we have so far been told is soon, which in terms of the EVE Online developers could be anywhere from tomorrow to some point after the heat death of the universe.